### PR TITLE
gh-87390: Fix starred tuple equality and pickling (alt)

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
It is a simpler alternative of #92249.

#87390

### CODE REVIEW

File: Lib/test/test_genericalias.py [hunk 1]
```diff
@@ -394,6 +396,7 @@ def test_pickle(self):
                     self.assertEqual(loaded.__origin__, alias.__origin__)
                     self.assertEqual(loaded.__args__, alias.__args__)
                     self.assertEqual(loaded.__parameters__, alias.__parameters__)
+                    self.assertEqual(type(loaded), type(alias))
```
Comment (JelleZijlstra) [inline Lib/test/test_genericalias.py:399]: maybe assertIs?

**Review by JelleZijlstra — COMMENTED** [review]
Thanks, this is indeed a better solution.

Could you add tests for pickling starred aliases though? Your new tests only cover equality.

**Comment by Fidget-Spinner** [comment]
LGTM. But IMO this needs a news, and like Jelle mentioned more tests (maybe we can take the ones from Mathew's PR and credit him?)

**Comment by serhiy-storchaka** [comment]
Additional tests in Mathew's PR are for the third parameter of GenericAlias. That change is not included in this PR.

File: Lib/test/test_genericalias.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/test/test_genericalias.py:399]: It is the same for types. And I do not want to restart the CI testing for such minor change. :wink:

**Comment by mrahtz** [comment]
Thanks Serhiy!

**Pull request merged**